lib/diagnostic.rb

```diff
@@ -43,6 +46,10 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open(filename) do |file|
+    file.each do |line|
+      
+      #replace trailing new line char "\n" with " "
+    end
```

Comment:
`File.open(filename)` is correct. However, the input parameter to the block is actually a single line of text from the file, rather than the entire contents.